src/Contacts/PhoneNumber.m

```diff
@@ -115,6 +119,75 @@ + (PhoneNumber *)tryParsePhoneNumberFromUserSpecifiedText:(NSString *)text {
     return [self phoneNumberFromUserSpecifiedText:sanitizedString];
 }
 
++ (NSArray *)tryParsePhoneNumbersFromsUserSpecifiedText:(NSString *)text
+                                      clientPhoneNumber:(NSString *)clientPhoneNumber {
+    assert(text != nil);
+    
+    text = [text stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+    if ([text isEqualToString:@""]) {
+        return nil;
+    }
+    
+    NSString *sanitizedString = [self removeFormattingCharacters:text];
+    assert(sanitizedString != nil);
+    
+    NSMutableArray *result = [NSMutableArray new];
+    NSMutableSet *phoneNumberSet = [NSMutableSet new];
+    void (^tryParsingWithCountryCode)(NSString *, NSString *) = ^(NSString *text,
+                                                      NSString *countryCode) {
+        PhoneNumber *phoneNumber = [PhoneNumber phoneNumberFromText:text
+                                                          andRegion:countryCode];
+        if (phoneNumber &&
+            ![phoneNumberSet containsObject:[phoneNumber toE164]]) {
```

Comment:
We can't used a dict because we want to return the `PhoneNumber`s and the ordering of the results matters. a) we want the ordering to be stable to provide a predictable UX b) we want "better" matches to appear earlier in the results.

I've added a check that `toE164` is non-nil.